packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -48,8 +48,12 @@ const ProductDetail = ({category, product, isLoading}) => {
     const toast = useToast()
     const navigate = useNavigation()
     const [primaryCategory, setPrimaryCategory] = useState(category)
+    const [productSetSelection, setProductSetSelection] = useState({})
+    const setProductsRefs = React.useRef({})
```

Comment:
The way I am able to keep track of what products in a set are selected is by creating a state object to map the products that are selected. Items are added by means of a new property on the product view component called `variantSelected`. This function is passes the variant that is selected, we then put it in the state object keyed on it's id. 

I use this information to determine what products aren't selected when you click the 'add all to cart' button and scroll to that components ref.